src/scripts.js

```diff
@@ -41,47 +40,69 @@ function displayAllRecipeCards(allRecipeData) {
+  return dataSet[Math.floor(Math.random() * dataSet.length)]
+}
 function searchByIngrients() {
   const searchResultRecipes = recipesRepo.filterRecipesByIngredients(ingredientsData, searchInput.value);
   displayAllRecipeCards({ recipes: searchResultRecipes });
 }
-
 function searchByRecipeName() {
   const searchResultName = recipesRepo.filterRecipesByName(searchInput.value);
   displayAllRecipeCards({ recipes: [searchResultName] });
 }
+function searchFavoriteRecipesByName() {
+  const searchResultName = currentUser.filterFavoritesByName(searchInput.value);
+  displayAllRecipeCards({ recipes: [searchResultName] });
+}
+function searchFavoriteRecipesByIngredient() {
+  const searchResultRecipes = currentUser.filterFavoritesByIngredients(ingredientsData, searchInput.value);
+  console.log(searchResultRecipes);
+  displayAllRecipeCards({ recipes: searchResultRecipes });
+}
 
 function handleSearchDropDown(event) {
   event.preventDefault();
   let searchBy = document.getElementById('search-recipe-select').value;
-  if (searchBy === 'recipe') {
-    searchByRecipeName(event);
-  } else if (searchBy === 'ingredient') {
-    searchByIngrients(event);
+  if (searchBy === 'recipe' && pageTitleText.innerText === 'Whats Cookin') {
+    searchByRecipeName();
+  } else if (searchBy === 'ingredient' && pageTitleText.innerText === 'Whats Cookin') {
+    searchByIngrients();
+  } else if (searchBy === 'recipe' && pageTitleText.innerText === 'Favorite Recipes') {
+    searchFavoriteRecipesByName();
+  } else if (searchBy === 'ingredient' && pageTitleText.innerText === 'Favorite Recipes') {
+    searchFavoriteRecipesByIngredient();
   }
   searchInput.value = '';
 }
-
 function displayPageLoad() {
   displayAllRecipeCards(recipesRepo);
+  createNewUser();
+  // createUniqueTags(recipesRepo);
+  console.log(currentUser);
 }
-
 function filterRecipesByTags(event) {
   const filteredRecipes = recipesRepo.filterRecipesByTag(event.target.value);
-  if (event.target.value === 'all recipes') {
+  const filteredFavoriteRecipes = currentUser.filterFavoritesByTag(event.target.value);
+  if (event.target.value === 'all recipes' && pageTitleText.innerText === 'Whats Cookin') {
     displayAllRecipeCards(recipesRepo);
-  } else if (event.target.value) {
+  } else if (pageTitleText.innerText === 'Whats Cookin') {
     displayAllRecipeCards({ recipes: filteredRecipes });
+  } else if (event.target.value === 'all recipes' && pageTitleText.innerText === 'Favorite Recipes') {
+    displayAllRecipeCards({ recipes: currentUser.favoriteRecipes });
+  } else if (pageTitleText.innerText === 'Favorite Recipes') {
+    displayAllRecipeCards({ recipes: filteredFavoriteRecipes });
   }
 }
-
 function handleRecipeClick(event) {
   const recipeId = Number(event.target.closest('.recipe-img-container').children[0].id);
   currentRecipe = recipesRepo.recipes.find(recipe => recipe.id === recipeId);
   displaySingleRecipe(currentRecipe);
   displayCostOfRecipe(currentRecipe);
-
```

Comment:
I don't think we should be removing the space between the functions for readability